src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -135,15 +136,32 @@ trait Implicits extends splain.SplainData {
         }
       }
       if (settings.lintImplicitRecursion) {
-        val s = if (rts.isAccessor) rts.accessed else if (rts.isModule) rts.moduleClass else rts
-        if (s != NoSymbol && context.owner.hasTransOwner(s))
-          context.warning(result.tree.pos, s"Implicit resolves to enclosing $rts", WarningCategory.WFlagSelfImplicit)
+        val s =
+          if (rts.isAccessor) rts.accessed
+          else if (rts.isModule) rts.moduleClass
+          else rts
+        val rtsIsImplicitWrapper = isView && rts.isMethod && rts.isSynthetic && rts.isImplicit
+        def isSelfEnrichment(encl: Symbol): Boolean =
+          tree.symbol.isParamAccessor && tree.symbol.owner == encl && !encl.isDerivedValueClass
```

Comment:
`SelfEnrichment` is a pun for our times.

Reply:
😢